src/utils/seedsApi.ts

```diff
@@ -0,0 +1,145 @@
+import { isValidChecksumAddress } from 'ethereumjs-util'
+import { Request, Response, NextFunction } from 'express'
+import _ from 'lodash'
+import Web3 from 'web3';
+
+import { Table } from '../db/db';
+import db from '../db/sql-db'
+import { getCrdtUpdateMessage } from '../types/message'
+import { NFTContractTypeName, NftFactory, NFTStandard } from '../types/nft';
+import { MusicPlatform, MusicPlatformType } from '../types/platform';
+
+type SeedEntity = 'platform' | 'contract'
+
+enum SeedPlatformRequiredKeys {
+  ID = 'id',
+  NAME = 'name',
+  TYPE = 'type',
+}
+
+enum SeedNftFactoryRequiredKeys {
+  ID = 'id',
+  // startingBlock?: string, // ignore optional
+  PLATFORM_ID = 'platformId',
+  CONTRACT_TYPE = 'contractType', // validated against `NFTContractTypeName`
+  // name?: string, // ignore optional
+  // symbol?: string, // ignore optional
+  // typeMetadata?: TypeMetadata // ignore optional
+  STANDARD = 'standard', // validated against `NFTStandard`
+  AUTO_APPROVE = 'autoApprove',
+  APPROVED = 'approved'
+}
+
+type SeedPayload = {
+  entity: SeedEntity,
+  data: any
+}
+
+export const authMiddleware = (
+  request: Request,
+  response: Response,
+  next: NextFunction,
+) => {
+
+  try {
+    let signer: any;
+
+    const auth = {
+      message: JSON.stringify(request.body),
+      signature: request.header('x-signature') || '',
+    }
+
+    try {
+      signer = validateSignature(auth, permittedAdminAddresses());
+    } catch (e) {
+      throw e;
+    }
+
+    next();
+  } catch (e) {
+    response.status(403).send('Authentication failed');
+    return;
+  }
+};
+
+export function validateSignature(signatureData: {
+  message: string;
+  signature: string;
+}, permittedAddresses: string[]): string {
+  let signer: string;
+
+  try {
+    const web3 = new Web3();
+    signer = web3.eth.accounts.recover(
+      signatureData.message,
+      signatureData.signature,
+    );
+  } catch (e) {
+    throw `Error verifying signature: ${e}`;
+  }
+
+  if (
+    !isValidChecksumAddress(signer) ||
+    !permittedAddresses.includes(signer.toLowerCase())
+  ) {
+    throw `Invalid signer address: ${signer}`;
+  }
+
+  return signer;
+}
+
+const permittedAdminAddresses = (): string[] => {
+  const addresses = process.env.PERMITTED_ADMIN_ADDRESSES?.toLowerCase();
+  if (!addresses) {
+    throw new Error('PERMITTED_ADMIN_ADDRESSES not set');
+  }
+  return addresses.split(',');
+}
+
+
+export const validateSeed = (payload: SeedPayload) => {
+  if (payload.entity === 'platform') {
```

Comment:
maybe can make these dynamic too, with higher order functions ie something like:
```suggestion

const typeValidator = (type:any, key: string) => (data: any) => {
  return type => Object.values(type).includes(data[key])
}

  const validPlatformType = typeValidator(MusicPlatformType, 'type')

  const validatorFunctions = {
  'platform': [minimumKeysPresent, validPlatformType],
  'nftFactory': [minimumKeysPresent, validContractType, validNFTStandard],
  }
  const validators = validatorFunctions[payload.entity];
  validators.forEach(validator => validator(payload.data));
  ```

Reply:
This is nice, will implement at the end as cleanup